static/js/tools/map/chart.tsx

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Chart component for drawing a choropleth.
+ */
+
+import React, { useEffect } from "react";
+import { GeoJsonData, GeoJsonFeatureProperties } from "../../chart/types";
+import { PlaceInfo, StatVarInfo } from "./context";
+import { Container } from "reactstrap";
+import _ from "lodash";
+import { drawChoropleth } from "../../chart/draw_choropleth";
+import {
+  updateHashPlaceInfo,
+  updateHashStatVarInfo,
+  USA_CHILD_PLACE_TYPES,
+} from "./util";
+import { urlToDomain } from "../../shared/util";
+
+interface ChartProps {
+  geoJsonData: GeoJsonData;
+  dataValues: { [dcid: string]: number };
+  placeInfo: PlaceInfo;
+  statVarInfo: StatVarInfo;
+  statVarDates: { [dcid: string]: string };
+  sources: Set<string>;
+}
+
+const SVG_CONTAINER_ID = "choropleth_map";
+const MAP_REDIRECT_PREFIX = "/tools/map";
+
+export function Chart(props: ChartProps): JSX.Element {
+  useEffect(() => {
+    draw(props);
+  }, [props]);
+  const title = getTitle(
+    Object.values(props.statVarDates),
+    props.statVarInfo.name
+  );
+  const sourcesJsx = getSourcesJsx(props.sources);
+  const placeDcid = props.placeInfo.enclosingPlace.dcid;
+  const statVarDcid = _.findKey(props.statVarInfo.statVar);
+  return (
+    <div>
+      <Container>
+        <div className="chart-section">
+          <div className="map-title">
+            <h3>{title}</h3>
+          </div>
+          <div id={SVG_CONTAINER_ID}></div>
+          <div className="map-footer">
+            <div className="sources">Data from {sourcesJsx}</div>
+            <div
+              className="explore-timeline-link"
+              onClick={() => exploreTimelineOnClick(placeDcid, statVarDcid)}
+            >
+              <span className="explore-timeline-text">Explore timeline</span>
+              <i className="material-icons">keyboard_arrow_right</i>
+            </div>
+          </div>
+        </div>
+      </Container>
+    </div>
+  );
+}
+
+function draw(props: ChartProps): void {
+  document.getElementById(SVG_CONTAINER_ID).innerHTML = "";
+  const width = document.getElementById(SVG_CONTAINER_ID).offsetWidth;
+  const height = (width * 2) / 5;
```

Comment:
I don't think the element has any height before the chart is rendered .. and I don't think we want to set a height for the div in the css because we might want different heights for different window sizes?